src/modules/navigator/gpsfailure.cpp

```diff
@@ -83,9 +83,22 @@ GpsFailure::on_active()
 	case GPSF_STATE_LOITER: {
 			/* Position controller does not run in this mode:
 			 * navigator has to publish an attitude setpoint */
+
+			struct position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
+
 			vehicle_attitude_setpoint_s att_sp = {};
 			att_sp.timestamp = hrt_absolute_time();
-			att_sp.roll_body = math::radians(_param_nav_gpsf_r.get());
+
+			//This determines if the fixedwing vehicle is landing or not, and if so it forces the wings level
+			//Otherwise the plane goes to the parameter set fixed bank loiter setting
+
+			if (pos_sp_triplet->current.valid && pos_sp_triplet->current.type == position_setpoint_s::SETPOINT_TYPE_LAND) {
+				att_sp.roll_body = 0.0f;
+
+			} else {
+				att_sp.roll_body = math::radians(_param_nav_gpsf_r.get());
+			}
+
 			att_sp.pitch_body = math::radians(_param_nav_gpsf_p.get());
 			att_sp.thrust_body[0] = _param_nav_gpsf_tr.get();
```

Comment:
Does it still make sense to use the same gps failsafe throttle? This might result in a fly away if misconfigured

Reply:
Good point!  Should we also force throttle to 0 if already landing?

Reply:
After thinking about it more I think we should also force throttle to 0.  If the vehicle was already landing, the intent is to get on the ground.  In this case the best course of action is to continue wings level and cut throttle to ensure the vehicle gets on the ground.  @Kjkinney can you add this code and code comments explaining why?